code-challenges.js

```diff
@@ -2,35 +2,155 @@
 
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
-
-
-
-
 var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
-// --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
 
+/**
+* VERSION 1.0
+*/
+
+// use a switch for each scenario of vowels (except u)
+// iterate through string; try using map
+// if letter is one of the 4 vowels, convert it accordingly
+// if letter is not one of the 4 vowels, then return it as is and move on
+
+const charConversion = str => {
+  // change all the letters to lowercase to ensure matching of the cases
+  // must convert string to array to use map by using split()
+  // break is optional and used with each case; once the match is found, it'll exit switch
+  // if break is omitted, the program continues execution at the next statement
+  return str.toLowerCase().split("").map(char => {
+    switch (char) {
+      case 'a':
+        return "4";
+      break;
+      case 'e':
+        return "3";
+      break;
+      case 'i':
+        return "1";
+      break;
+      case 'o':
+        return "0";
+      break;
+      default:
+        return char;
+    }
+  }).join("");
+}
+
+console.log(charConversion(secretCodeWord1));
+console.log(charConversion(secretCodeWord2));
+
+
+/**
+  * VERSION 2.0
+  */
+
+const charConversionV2 = str => {
+  // store each unique key with its corresponding value
+  const chars = {"a": 4, "e": 3, "i": 1, "o": 0};
+  // replace() looks for a match using regex
+  // the anonymous function takes the matching letter and pass it to the chars object to access the value of that specific letter through bracket notation
+  // that value then replaces the matched letter thanks to the replace() method :)
+  return str.replace(/[aeio]/gi, letter => chars[letter])
+}
+
+console.log(charConversionV2(secretCodeWord1));
+console.log(charConversionV2(secretCodeWord2));
```

Comment:
thank you for showing both approaches! i like both of them. and especially thank you for talking throught he logic of your regex :)